scripts/bin/auto-mount

```diff
@@ -1,17 +1,50 @@
-#! /bin/sh
-
-sleep 5
-
-$SNAP/usr/local/bin/udisksctl monitor | grep --line-buffered " Device:" | while read -r -- device; do
-
-  echo "$SNAP_NAME.auto-mount found ${device}"
-  if echo $device| grep -q "[0-9]$"; then
-    DEV="$(echo $device | sed 's/^Device: //')"
-  fi
-  if [ -n "$DEV" ]; then
-    RES=`$SNAP/usr/local/bin/udisksctl mount --no-user-interaction --block-device "$DEV"`
-    # process returned value into the mount point string
-    RES2=`echo $RES | cut -f4 -d\  | cut -f1 -d.`
-    echo $RES2 > $WATCHED
-  fi
-done
+#!/bin/sh
+
+. "$SNAP/bin/common"
+
+watch_devices() {
+    inotifywait --quiet --monitor --event "moved_to" --event "delete" /dev/disk/by-uuid | \
+        while read -r _parent _action _uuid; do
+            case "$_action" in
+                MOVED_TO) try_mount "$_parent" "$_uuid" ;;
+                DELETE)   rm_mount  "$_uuid" ;;
+                *) : ;;
+            esac
+        done
+}
+
+try_mount() {
+    _parent="${1%/}"
```

Comment:
Odd nit: technically we don't have to do a substitution to strip the lagging `/` because (relevant enough) versions of libc collapse particular repetitions of `/` to the expected result. I believe in all cases, `//` is collapsed to `/`. 

I think you only did this to make the code nicer to look at, and I *doubt* there's any actual problem with doing this, but I felt like showing off.